app/lib/git_repository.rb

```diff
@@ -585,7 +585,7 @@ def changes_at_path?(path)
 
   # Gets all entries at directory +path+ of a specified +type+ (:blob, :tree, or nil for both), as
   # Repository::RevisionFile and Repository::RevisionDirectory.
-  def entries_at_path(path, type=nil)
+  def entries_at_path(path, type=nil, ignore_internal=true)
```

Comment:
I think you should filter the .gitkeep files externally, instead of making the api more complex.
(i.e. get all the entries, then drop the ones you don't need)